addon/wait-for-promise.ts

```diff
@@ -0,0 +1,45 @@
+import { DEBUG } from '@glimmer/env';
+import TestWaiter from './test-waiter';
+
+const PROMISE_WAITER = new TestWaiter('promise-waiter');
+
+/**
+ * Allows for tracking the beginning and end of async operations within promises.
```

Comment:
I don't think this is quite correct. How does this sound:

```suggestion
  A convenient utility function to simplify waiting for a promise.
```